app/models/group.rb

```diff
@@ -10,25 +10,64 @@
 class Group < ApplicationRecord
   belongs_to :leader, class_name: 'User'
   belongs_to :draw
-  has_many :members, class_name: 'User'
+  has_many :memberships
+  has_many :members, through: :memberships, source: :user
 
-  validates :size, presence: true, numericality: { only_integer: true,
-                                                   greater_than: 0 }
-  validates :status, presence: true
-  validates :leader, presence: true
+  enum status: %w(open full locked)
+
+  validates :size, presence: true,
+                   inclusion: { in: ->(group) { group.draw.suite_sizes },
+                                if: proc { |g| g.draw.present? } }
+  validates :status, presence: true, inclusion: { in: Group.statuses.keys }
+  validates :leader, presence: true,
+                     inclusion: { in: ->(group) { group.members } }
   validates :draw, presence: true
 
-  enum status: %w(open full locked)
+  # validate :validate_members_count, if: proc { |g| g.size.present? }
+  validate :validate_status, if: proc { |g| g.size.present? }
 
-  after_save :add_leader_to_members
+  before_validation :add_leader_to_members, if: proc { |g| g.leader.present? }
 
   def name
     "#{leader.name}'s Group"
   end
 
+  def update_status
```

Comment:
Right, but should we maybe prevent the modification of a membership's group or user following creation? I feel like we could do that with a simple validation to make sure that neither of those associations are ever modified except during create. What do you think?